extensions/okhttp/src/main/java/com/google/android/exoplayer2/ext/okhttp/OkHttpDataSource.java

```diff
@@ -304,14 +304,14 @@ public long open(DataSpec dataSpec) throws HttpDataSourceException {
 
     // Check for a valid response code.
     if (!response.isSuccessful()) {
+      Map<String, List<String>> headers = response.headers().toMultimap();
       byte[] errorResponseBody;
       try {
         errorResponseBody = Util.toByteArray(Assertions.checkNotNull(responseByteStream));
       } catch (IOException e) {
-        throw new HttpDataSourceException(
-            "Error reading non-2xx response body", e, dataSpec, HttpDataSourceException.TYPE_OPEN);
+        throw new InvalidResponseCodeException(
+            responseCode, response.message(), headers, dataSpec, null);
```

Comment:
The `responseBody` argument isn't `@Nullable`, so you should use `Util.EMPTY_BYTE_ARRAY` as suggested above. Alternatively, you'll need to add `@Nullable` annotations through `InvalidResponseCodeException`. I think it's probably best to use `Util.EMPTY_BYTE_ARRAY` unless you can think of a case where it's important to be able to disambiguate between the cases described above.